examples/mnist/assets/algo_cnn/algo.py

```diff
@@ -0,0 +1,91 @@
+import substratools as tools
+import os
+import keras
+from keras import backend as K
+from keras.models import Sequential
+from keras.layers import Dense, Dropout, Flatten
+from keras.layers import Conv2D, MaxPooling2D
+
+# input image dimensions
+img_rows, img_cols = 28, 28
+input_shape = (img_rows, img_cols, 1)
+
+
+class Algo(tools.algo.Algo):
+    def _normalize_X(self, X):
+
+        if K.image_data_format() == "channels_first":
+            X = X.reshape(X.shape[0], 1, img_rows, img_cols)
+
+            input_shape = (1, img_rows, img_cols)
+        else:
+            X = X.reshape(X.shape[0], img_rows, img_cols, 1)
+
+            input_shape = (img_rows, img_cols, 1)
+
+        X = X.astype("float32")
+
+        X /= 255
+
+        print(f"X shape: {X.shape}")
+        print(f"{X.shape[0]} samples")
+
+        return X
+
+    def train(self, X, y, models, rank):
+        X = self._normalize_X(X)
+        num_classes = 10
```

Comment:
Couldn't we get that from `y` instead?

Reply:
I think `y` is a list of Integer and not a list of vector (that's why we use `to_categorical` to transform it into a vector), so maybe we could try to get the max on y but i think it's overkill. We will always have 10 digits on MNIST